app/questionnaire/value_source_resolver.py

```diff
@@ -162,25 +176,27 @@ def resolve(
     ) -> Union[ValueSourceEscapedTypes, ValueSourceTypes]:
         source = value_source["source"]
 
-        if source == "answers":
-            return self._resolve_answer_value_source(value_source)
-
-        if source == "list":
-            return self._resolve_list_value_source(value_source)
-
-        if source == "metadata":
+        def resolve_metadata_source(_value_source):
             if not self.metadata:
                 raise NoMetadataException
-            identifier = value_source["identifier"]
+            identifier = _value_source["identifier"]
             return self.metadata[identifier]
 
-        if source == "location" and value_source.get("identifier") == "list_item_id":
-            # This does not use the location object because
-            # routes such as individual response does not have the concept of location.
-            return self.list_item_id
+        def resolve_location_source(_value_source):
+            if _value_source.get("identifier") == "list_item_id":
+                return self.list_item_id
+
+        def resolve_repsonse_metadata_source(_value_source):
+            return self.response_metadata.get(_value_source.get("identifier"))
 
-        if source == "response_metadata":
-            return self.response_metadata.get(value_source.get("identifier"))
+        resolve_method_mapping = {
+            "answers": self._resolve_answer_value_source,
+            "list": self._resolve_list_value_source,
+            "metadata": resolve_metadata_source,
+            "location": resolve_location_source,
+            "response_metadata": resolve_repsonse_metadata_source,
+            "calculated_summary": self._resolve_calculated_summary_value_source,
+            "progress": self._resolve_progress_value_source,
```

Comment:
FYI: the "progress" resolver is the only bit I've added to the logic.

I've had to refactor the function to a mapping pattern instead of the if conditions, as PyLint was complaining of too many return statements.